test/controllers/user_roles_controller_test.rb

```diff
@@ -134,5 +134,9 @@ def test_revoke
     end
     assert_redirected_to user_path(target_user.display_name)
     assert_equal "The string `no_such_role' is not a valid role.", flash[:error]
+
+    # Revoking administrator role from current user should fail
+    post :revoke, :params => { :display_name => administrator_user.display_name, :role => "administrator" }
+    assert_redirected_to user_path(administrator_user.display_name)
```

Comment:
Can we assert that the flash message is shown here... I think there some examples of doing it in other tests.

Reply:
I added the flash msg assert as well now.

Reply:
Travis CI reports some spurious issue, which doesn't appear related to this PR in any way. Local test was ok, though (with a different seed value).

Reply:
If a test with a different seed value is ok it means there is some dependency between tests so that they only work when run in a certain order.

Reply:
Actually the problem is that test tries to fetch an arbitrary node ID expecting it not to exist so fails if a previous test happens to have created it.